### ISSUE TITLE
Migration Crash from 151 to 153

### ISSUE THREAD

**Issue by Belphemur** [zadam/trilium#717]
Version : 0.37.3

```
Nov 22 17:46:04 dolma.hole trilium[2026]: App HTTP server starting up at port 8080
Nov 22 17:46:04 dolma.hole trilium[2026]: {
Nov 22 17:46:04 dolma.hole trilium[2026]:   "appVersion": "0.37.3",
Nov 22 17:46:04 dolma.hole trilium[2026]:   "dbVersion": 153,
Nov 22 17:46:04 dolma.hole trilium[2026]:   "syncVersion": 11,
Nov 22 17:46:04 dolma.hole trilium[2026]:   "buildDate": "2019-11-19T23:05:54+01:00",
Nov 22 17:46:04 dolma.hole trilium[2026]:   "buildRevision": "07043fb177afb9d754428a410b3019d53d7b6fa0",
Nov 22 17:46:04 dolma.hole trilium[2026]:   "dataDirectory": "/srv/dev-disk-by-label-Documents/Docs/Trilium/",
Nov 22 17:46:04 dolma.hole trilium[2026]:   "clipperProtocolVersion": "1.0"
Nov 22 17:46:04 dolma.hole trilium[2026]: }
Nov 22 17:46:04 dolma.hole trilium[2026]: App db version is 153, while db version is 151. Migration needed.
Nov 22 17:46:04 dolma.hole trilium[2026]: Created backup at /srv/dev-disk-by-label-Documents/Docs/Trilium//backup/backup-before-migration.db
Nov 22 17:46:04 dolma.hole trilium[2026]: Attempting migration to version 152
Nov 22 17:46:04 dolma.hole trilium[2026]: Migration with SQL script: CREATE TABLE IF NOT EXISTS "notes_mig" (
Nov 22 17:46:04 dolma.hole trilium[2026]:                                            `noteId`        TEXT NOT NULL,
Nov 22 17:46:04 dolma.hole trilium[2026]:                                            `title`        TEXT NOT NULL DEFAULT "note",
Nov 22 17:46:04 dolma.hole trilium[2026]:                                            `contentLength`        INT NOT NULL,
Nov 22 17:46:04 dolma.hole trilium[2026]:                                            `isProtected`        INT NOT NULL DEFAULT 0,
Nov 22 17:46:04 dolma.hole trilium[2026]:                                            `type` TEXT NOT NULL DEFAULT 'text',
Nov 22 17:46:04 dolma.hole trilium[2026]:                                            `mime` TEXT NOT NULL DEFAULT 'text/html',
Nov 22 17:46:04 dolma.hole trilium[2026]:                                            `hash` TEXT DEFAULT "" NOT NULL,
Nov 22 17:46:04 dolma.hole trilium[2026]:                                            `isDeleted`        INT NOT NULL DEFAULT 0,
Nov 22 17:46:04 dolma.hole trilium[2026]:                                            `isErased`        INT NOT NULL DEFAULT 0,
Nov 22 17:46:04 dolma.hole trilium[2026]:                                            `dateCreated`        TEXT NOT NULL,
Nov 22 17:46:04 dolma.hole trilium[2026]:                                            `dateModified`        TEXT NOT NULL,
Nov 22 17:46:04 dolma.hole trilium[2026]:                                            `utcDateCreated`        TEXT NOT NULL,
Nov 22 17:46:04 dolma.hole trilium[2026]:                                            `utcDateModified`        TEXT NOT NULL,
Nov 22 17:46:04 dolma.hole trilium[2026]:                                            PRIMARY KEY(`noteId`));
Nov 22 17:46:04 dolma.hole trilium[2026]: INSERT INTO notes_mig (noteId, title, contentLength, isProtected, type, mime, hash, isDeleted, isErased, dateCreated, dateModified, utcDateCreated, utcDateModified)
Nov 22 17:46:04 dolma.hole trilium[2026]: SELECT noteId, title, -1, isProtected, type, mime, hash, isDeleted, isErased, dateCreated, dateModified, utcDateCreated, utcDateModified FROM notes;
Nov 22 17:46:04 dolma.hole trilium[2026]: DROP TABLE notes;
Nov 22 17:46:04 dolma.hole trilium[2026]: ALTER TABLE notes_mig RENAME TO notes;
Nov 22 17:46:04 dolma.hole trilium[2026]: UPDATE notes SET contentLength = (SELECT COALESCE(LENGTH(content), 0) FROM note_contents WHERE note_contents.noteId = notes.note$d);
Nov 22 17:46:04 dolma.hole trilium[2026]: CREATE INDEX `IDX_notes_isDeleted` ON `notes` (`isDeleted`);
Nov 22 17:46:04 dolma.hole trilium[2026]: CREATE INDEX `IDX_notes_title` ON `notes` (`title`);
Nov 22 17:46:04 dolma.hole trilium[2026]: CREATE INDEX `IDX_notes_type` ON `notes` (`type`);
Nov 22 17:46:04 dolma.hole trilium[2026]: CREATE INDEX `IDX_notes_dateCreated` ON `notes` (`dateCreated`);
Nov 22 17:46:04 dolma.hole trilium[2026]: CREATE INDEX `IDX_notes_dateModified` ON `notes` (`dateModified`);
Nov 22 17:46:04 dolma.hole trilium[2026]: CREATE INDEX `IDX_notes_utcDateModified` ON `notes` (`utcDateModified`);
Nov 22 17:46:04 dolma.hole trilium[2026]: CREATE INDEX `IDX_notes_utcDateCreated` ON `notes` (`utcDateCreated`);
Nov 22 17:46:04 dolma.hole trilium[2026]: -- should be OK since sync protocol changes so all instances must upgrade
Nov 22 17:46:04 dolma.hole trilium[2026]: UPDATE attributes SET isDeleted = 1 WHERE name = 'fileSize';
Nov 22 17:46:04 dolma.hole trilium[2026]: Trace: ERROR: Error executing query. Inner exception: Error: SQLITE_CONSTRAINT: NOT NULL constraint failed: notes.contentLengthE$ror
Nov 22 22 17:46:04 dolma.hole trilium[2026]:     at wrap (/srv/dev-disk-by-label-Data/syncthing/trilium/src/services/sql.js:165:23)
Nov 22 17:46:04 dolma.hole trilium[2026]:     at Object.executeScript (/srv/dev-disk-by-label-Data/syncthing/trilium/src/services/sql.js:161:18)
Nov 22 17:46:04 dolma.hole trilium[2026]:     at /srv/dev-disk-by-label-Data/syncthing/trilium/src/services/migration.js:55:31
Nov 22 17:46:04 dolma.hole trilium[2026]:     at /srv/dev-disk-by-label-Data/syncthing/trilium/src/services/sql.js:201:25
Nov 22 17:46:04 dolma.hole trilium[2026]:     at Object.error (/srv/dev-disk-by-label-Data/syncthing/trilium/src/services/log.js:33:13)
Nov 22 17:46:04 dolma.hole trilium[2026]:     at wrap (/srv/dev-disk-by-label-Data/syncthing/trilium/src/services/sql.js:171:13)
Nov 22 17:46:04 dolma.hole trilium[2026]: Trace: ERROR: Error executing transaction, executing rollback. Inner stack: Error
Nov 22 17:46:04 dolma.hole trilium[2026]:     at wrap (/srv/dev-disk-by-label-Data/syncthing/trilium/src/services/sql.js:165:23)
Nov 22 17:46:04 dolma.hole trilium[2026]:     at Object.executeScript (/srv/dev-disk-by-label-Data/syncthing/trilium/src/services/sql.js:161:18)
Nov 22 17:46:04 dolma.hole trilium[2026]:     at /srv/dev-disk-by-label-Data/syncthing/trilium/src/services/migration.js:55:31
Nov 22 17:46:04 dolma.hole trilium[2026]:     at /srv/dev-disk-by-label-Data/syncthing/trilium/src/services/sql.js:201:25
Nov 22 17:46:04 dolma.hole trilium[2026]: Outside stack: Error
Nov 22 17:46:04 dolma.hole trilium[2026]:     at Object.transactional (/srv/dev-disk-by-label-Data/syncthing/trilium/src/services/sql.js:192:23)
Nov 22 17:46:04 dolma.hole trilium[2026]:     at Object.migrate (/srv/dev-disk-by-label-Data/syncthing/trilium/src/services/migration.js:49:23)
Nov 22 17:46:04 dolma.hole trilium[2026]:     at async /srv/dev-disk-by-label-Data/syncthing/trilium/src/services/sql_init.js:74:13
Nov 22 17:46:04 dolma.hole trilium[2026]:     at async Object.init (/srv/dev-disk-by-label-Data/syncthing/trilium/src/services/cls.js:5:12)
Nov 22 17:46:04 dolma.hole trilium[2026]:     at async initDbConnection (/srv/dev-disk-by-label-Data/syncthing/trilium/src/services/sql_init.js:53:5)
Nov 22 17:46:04 dolma.hole trilium[2026]:     at Object.error (/srv/dev-disk-by-label-Data/syncthing/trilium/src/services/log.js:33:13)
Nov 22 17:46:04 dolma.hole trilium[2026]:     at /srv/dev-disk-by-label-Data/syncthing/trilium/src/services/sql.js:210:21
Nov 22 17:46:04 dolma.hole trilium[2026]: Trace: ERROR: error during migration to version 152: Error
Nov 22 17:46:04 dolma.hole trilium[2026]:     at wrap (/srv/dev-disk-by-label-Data/syncthing/trilium/src/services/sql.js:165:23)
Nov 22 17:46:04 dolma.hole trilium[2026]:     at Object.executeScript (/srv/dev-disk-by-label-Data/syncthing/trilium/src/services/sql.js:161:18)
Nov 22 17:46:04 dolma.hole trilium[2026]:     at /srv/dev-disk-by-label-Data/syncthing/trilium/src/services/migration.js:55:31
Nov 22 17:46:04 dolma.hole trilium[2026]:     at /srv/dev-disk-by-label-Data/syncthing/trilium/src/services/sql.js:201:25
Nov 22 17:46:04 dolma.hole trilium[2026]:     at Object.error (/srv/dev-disk-by-label-Data/syncthing/trilium/src/services/log.js:33:13)
Nov 22 17:46:04 dolma.hole trilium[2026]:     at Object.migrate (/srv/dev-disk-by-label-Data/syncthing/trilium/src/services/migration.js:74:17)
Nov 22 17:46:04 dolma.hole trilium[2026]: Trace: ERROR: migration failed, crashing hard
Nov 22 17:46:04 dolma.hole trilium[2026]:     at Object.error (/srv/dev-disk-by-label-Data/syncthing/trilium/src/services/log.js:33:13)
Nov 22 17:46:04 dolma.hole trilium[2026]:     at Object.migrate (/srv/dev-disk-by-label-Data/syncthing/trilium/src/services/migration.js:75:17)
Nov 22 17:46:04 dolma.hole systemd[1]: trilium.service: Main process exited, code=exited, status=1/FAILURE
Nov 22 17:46:04 dolma.hole systemd[1]: trilium.service: Failed with result 'exit-code'.
```

**Comment by zadam** [comment]
Hello, I fixed the issue in the migration script. This happens when there's a minor inconsistency in the database which should then be auto-fixed during the app startup.

**Comment by zadam** [comment]
Fix has been released in 0.37.5
